### DISCUSSION TITLE
CreateHandler not receiving request

### DISCUSSION THREAD

**Discussion by Shashikant-Kew** [graphql/graphql-http#131] [Q&A]
**Screenshot**
Visualising is always helpful.

**Expected Behaviour**
Like graphqlHttp of express-graphql, createHandler of this package should receive request object

**Actual Behaviour**
No request response is availalble in createHandler

**Debug Information**
Help us debug the bug?

**Further Information**
Anything else you might find helpful.

**Comment by enisdenjo** [comment]
graphql-http has thorough tests for each http framework's handlers ([see here](https://github.com/graphql/graphql-http/blob/main/tests/use.test.ts)). Receiving requests is the backbone of all tests - and they all pass.

There is a possibility there's something wrong with your setup or server instead. If you can share your server code and versions, or even better create a repro, I can take a look when I get some free time. Thanks!

**Reply by Shashikant-Kew** [reply]
can you please provide one example of create handler where context is modified or added with value from request object?
so context will have function returning object with that req field. Thanks
We are using express with node.

**Reply by Shashikant-Kew** [reply]
Thanks for the example but Issue is I am migrating from graphqlHttp of express-graphql to CreateHandler of graphql-http.
and in create handler it is just not getting request object. If I add statically something it gets assigned. But when I try to access headers it does not get reference.

app.all(
  "/graphql",
  createHandler({
    schema: schema,
    context: req => {
      profile: req.headers['user-det']
},
  })
)

here request not detected

even example given for context function not modifying/adding context var.
Please suggest.
Thanks.

**Reply by benjie** [reply]
The syntax of your JavaScript is incorrect; you have an arrow function that has a block body, with a labelled expression statement; or in other words your arrow function doesn’t return anything. You need to wrap the `{…}` in parentheses so that JavaScript interprets it as an object rather than a block.

**Reply by Shashikant-Kew** [reply]
app.all(
"/graphql",
createHandler({
schema: schema,
context: (req) => {
return {
profile: req.headers['user-det']
}
},
})
)

Please suggest if this is correct?

**Reply by enisdenjo** [reply]
That looks correct. There's too much moving parts now, if you would like us to help, please create a reproduction we can run and we'll investigate when we get some free time.
